holmes/core/toolset_manager.py

```diff
@@ -0,0 +1,314 @@
+        A toolset is considered a CLI tool if it has any of cli tool tags:
+        """
+        return [ToolsetTag.CORE, ToolsetTag.CLI]
+
+    @property
+    def server_tool_tags(self) -> List[str]:
+        """
+        Returns the list of toolset tags that are relevant for CLI tools.
+        A toolset is considered a CLI tool if it has any of UI tool tags:
+        """
+        return [ToolsetTag.CORE, ToolsetTag.CLUSTER]
+
+    @staticmethod
+    def get_toolset_definition_enabled(toolset_config: dict[str, Any]) -> bool:
+        """
+        Get the enabled status of the toolset from the config.
+        Return False only when the toolset is explicitly defined as disabled in the config.
+        This is helpful to enable the toolset specified in the config or custom toolset file without manually specify
+        'enabled: true'. Also, this avoid setting enabled to true by default, which can lead to unexpected behavior if
+        the toolset is not ready to be used.
+        """
+        if "enabled" in toolset_config:
+            if (
+                isinstance(toolset_config["enabled"], bool)
+                and not toolset_config["enabled"]
+            ):
+                return False
+            # Normally benedict should translate the value of enabled to bool when well assigned,
+            # but in case it doesn't, and to avoid unexpected behavior, we check if the value is a
+            # string and if it is 'false' (case insensitive).
+            elif (
+                isinstance(toolset_config["enabled"], str)
+                and toolset_config["enabled"].lower() == "false"
+            ):
+                return False
+            # TODO(mainred): add validation for the enabled field to be bool or str other than a valid 'bool' string
+        return True
+
+    def _list_all_toolsets(
+        self, dal: Optional[SupabaseDal] = None, check_prerequisites=True
+    ) -> List[Toolset]:
+        """
+        List all built-in and custom toolsets.
+        """
+        # Load built-in toolsets
+        builtin_toolsets = load_builtin_toolsets(dal)
+        toolsets_by_name: dict[str, Toolset] = {
+            toolset.name: toolset for toolset in builtin_toolsets
+        }
+
+        # build-in toolset is enabled when it's explicitly enabled in the config
+        if self.toolsets is not None:
+            # enabled built-in toolsets when they are defined not disabled in the config
+            for _, toolset_config in self.toolsets.items():
+                logging.debug(f"Processing toolset config: {toolset_config}")
+                toolset_config["enabled"] = self.get_toolset_definition_enabled(
+                    toolset_config
+                )
+            toolsets_from_config = load_toolsets_config(
+                toolsets=self.toolsets, strict_check=False
+            )
+            if toolsets_from_config:
+                self.add_or_merge_yaml_toolsets(
```

Comment:
There's some misunderstanding and less documented part for tool_config.
From the original code, both toolset and custom_toolset can override built-in toolset and introduce new customized toolset, which seems to be redundant.